sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -98,6 +102,8 @@ typedef struct _nib_onl_entry {
      * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0.
      */
     uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+    evtimer_msg_event_t reach_timeout;  /**< Event for @ref GNRC_IPV6_NIB_REACH_TIMEOUT */
+    evtimer_msg_event_t delay_timeout;  /**< Event for @ref GNRC_IPV6_NIB_DELAY_TIMEOUT */
```

Comment:
> And for routing 6LNs you probably need as much as you can get.

They are called 6LRs, but true, but experience shows that those should be less restricted nodes anyway with 6LoWPAN (and thus be rarer).

My count btw is ((`GNRC_IPV6_NIB_NUMOF` * 6) + (`GNRC_IPV6_NIB_OFFL_NUMOF` * 1) + (`GNRC_IPV6_NIB_DEFAULT_ROUTERS_NUMOF` * 2) + (`GNRC_NETIF_NUMOF` * 2) + `GNRC_IPV6_NIB_ABR_NUMOF` + 16) * 16 bytes = 848 bytes (with the configuration proposed in #7212 and considering one interface and one ABR). It can be expected that the number of bytes in the current implementation with a similar configuration is >1.5 kbytes.

These are the events listed by data structure (as documented in `net/gnrc/ipv6/nib.h`):

* On-link entries (of which there are `GNRC_IPV6_NIB_NUMOF` [4 in #7212]):
    1. `snd_ns` (handling both `SND_MC_NS` and `SND_UC_NS` event type, there is a type distinction since MC NS are rate-limited, so we need to look for future MC NS anytime we plan to send another)
    2. `snd_na` (handling `SND_NA` event type)
    3. *`reach_timeout`* (handling `REACH_TIMEOUT`)
    4. *`delay_timeout`* (handling `DELAY_TIMEOUT`) [this and the one above can potentially be merged]
    5. *`addr_reg_timeout`* (handling `ADDR_REG_TIMEOUT`)
    6. *`reply_rs`* (handling `REPLY_RS`)
* Off-link entries (of which there are `GNRC_IPV6_NIB_OFFL_NUMOF` [8 in #7212])
    1. `prefix_timeout` (I think the event type should now be clear)
* Default router entries (of which there are `GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF` [1 in #7212])
    1. `reconfirm_rtr`, this is currently documented wrong to be on on-link entries btw, will fix
    2. `rtr_timeout`, maybe this can be merged with `reconfirm_rtr` since a router timeout in my head only happen after at least one try of reconfirmation, but I need to have a look at the RFC for that and it's probably complicated to calculate the offset from n reconfirmation retries.
* Network interfaces (one assumed for a normal 6LR):
    1. `search_rtr` (I think the event type should now be clear)
    2. *`snd_mc_ra`* (since there are no multicast RA on 6LN, this can technically optimized out on pure 6LR)
* ABR (Authoritative Border Router, of which there is currently only expected to be one)
   1. *`abr_timeout`*
* 16 *`sixlo_ctx_timeout`* events for each possible context. These won't probably be real events though. Currently those are handled by just storing the timestamp of last update and being deprecated on next use if current timestamp greater than timestamp + timeout offset. I think I will keep it that way. The type in the model exists just for completeness.

With all possibly mentioned optimizations (merging the two state transition events, merging router reconfirmation and timeout and dropping the context timeouts) I get to 512 bytes for a 6LR, for a non-routing 6LN (which doesn't have NUD state-transitions and thus also doesn't really need more than one on-link entry (the one of the default router), doesn't have address registration timeouts, ABR timeouts and doesn't send RAs [thus not requiring `snd_mc_ra` and `reply_rs`]; I marked all the eligible cursive above) I even get down to 80 bytes with 1 on-link entry (for the default router and prefix), 1 off-link entry (for the prefix) and 1 default router.

All in all you *have* to remember that the 6Lo-ND optimization is (1) about power consumption (restricting multicast messages) and (2) assuming that 6LR are lower tier "power-houses" (after the 6LBR) in terms of RAM or non-existent (in a single-hop scenario among others e.g. with [IPv6 over BLE](https://tools.ietf.org/html/rfc7668#section-1.1)).